### ISSUE TITLE
Invesalius missing dependencies (Bugzilla Bug 2585)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/distribution#2585]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 2585
Date: 2020-01-11 14:21:29 +0000
From: daraf &lt;<angelob1@gmail.com>&gt;
To:   @berolinux
CC:   angelob1@gmail.com, @benbullard79, bugs@openmandriva.org, @itchka, nix.or.die@gmail.com, nobodydead@gmail.com, @rugyada, @tpgxyz

Last updated: 2021-05-23 03:20:36 +0000

## Comment 19988
Date: 2020-01-11 14:21:29 +0000
From: daraf &lt;<angelob1@gmail.com>&gt;

I opened the same bug for OMV-LX 3, and the advice was: AppImage.
But in OMV-4.0 neither AppImage works, so here I am, yet.
When I try installing Invesalius, I have this: 

['\n Problema: conflicting requests\n - nothing provides python-dicom needed by invesalius-3.0.1925-5.x86_64\n - nothing provides python-gdcm needed by invesalius-3.0.1925-5.x86_64\n - nothing provides python-itk needed by invesalius-3.0.1925-5.x86_64\n - nothing provides python-itk-numarray needed by invesalius-3.0.1925-5.x86_64\n - nothing provides python-nibabel needed by invesalius-3.0.1925-5.x86_64\n - nothing provides python-sigar needed by invesalius-3.0.1925-5.x86_64']

## Comment 19991
Date: 2020-01-11 16:48:33 +0000
From: @benbullard79

Confirmed.

There are 6 missing dependencies in OM Lx 4.1/Rolling.

# dnf install invesalius
Last metadata expiration check: 0:56:26 ago on Sat 11 Jan 2020 08:59:56 AM CST.
Error: 
 Problem: conflicting requests
  - nothing provides python-dicom needed by invesalius-3.0.1925-5.x86_64
  - nothing provides python-gdcm needed by invesalius-3.0.1925-5.x86_64
  - nothing provides python-itk needed by invesalius-3.0.1925-5.x86_64
  - nothing provides python-itk-numarray needed by invesalius-3.0.1925-5.x86_64
  - nothing provides python-nibabel needed by invesalius-3.0.1925-5.x86_64
  - nothing provides python-sigar needed by invesalius-3.0.1925-5.x86_64

Is probably the same in Cooker.

## Comment 20026
Date: 2020-02-08 12:57:36 +0000
From: daraf &lt;<angelob1@gmail.com>&gt;

It's so hard solving dependencies?

## Comment 20246
Date: 2020-05-21 02:19:58 +0000
From: @benbullard79

(In reply to daraf from comment #2)
> It's so hard solving dependencies?

It depends on the software and the people upstream that write or wrote the code. In the case of this package I'm told it is closer to a nightmare to get this package up to date with current OM Lx libraries. python in particular can make for some nightmarish dependency issues.

## Comment 20648
Date: 2021-05-23 02:55:22 +0000
From: @benbullard79

Moving to Cooker. As of today in Cooker:

# dnf install invesalius
...
Error: 
 Problem: conflicting requests
  - nothing provides python-dicom needed by invesalius-3.1.99993-1.x86_64
  - nothing provides python-gdcm needed by invesalius-3.1.99993-1.x86_64
  - nothing provides python-itk needed by invesalius-3.1.99993-1.x86_64
  - nothing provides python-itk-numarray needed by invesalius-3.1.99993-1.x86_64
  - nothing provides python-nibabel needed by invesalius-3.1.99993-1.x86_64
  - nothing provides python-sigar needed by invesalius-3.1.99993-1.x86_64

This software has no package maintainer. We need someone from the community to volunteer to maintain this package. Packages in unsupported repository are community supported.

## Comment 20649
Date: 2021-05-23 03:01:50 +0000
From: @benbullard79

I started builds for this in ABF for Cooker and Rolling.

## Comment 20654
Date: 2021-05-23 03:20:36 +0000
From: @benbullard79

Builds failed. Here are links to logs for cooker x86_64 and znver1 builds:

https://abf.openmandriva.org/build_lists/48554

https://abf.openmandriva.org/build_lists/48555

**Comment by mandian** [comment]
working on it but it requires bug #2696 fixed before.

**Comment by rugyada** [comment]
Closing, as no follow-up since long time.

**Comment by mandian** [comment]
invesalius 3.1.99997 is in repo and it is working.

**Comment by rugyada** [comment]
Good to read some feedback. Thanks

**Comment by benbullard79** [comment]
Testing invesalius in Lx 4.3. Installs but does not seem to open. Probably `ModuleNotFoundError: No module named 'invesalius_cy'`.

```
$ invesalius
Traceback (most recent call last):
  File "/usr/lib64/python3.9/site-packages/wx/core.py", line 2326, in Notify
    self.notify()
  File "/usr/lib64/python3.9/site-packages/wx/core.py", line 3534, in Notify
    self.result = self.callable(*self.args, **self.kwargs)
  File "/usr/share/invesalius/app.py", line 237, in Startup
    from invesalius.gui.frame import Frame
  File "/usr/share/invesalius/invesalius/gui/frame.py", line 29, in <module>
    import invesalius.gui.default_tasks as tasks
  File "/usr/share/invesalius/invesalius/gui/default_tasks.py", line 28, in <module>
    import invesalius.gui.data_notebook as nb
  File "/usr/share/invesalius/invesalius/gui/data_notebook.py", line 40, in <module>
    import invesalius.data.slice_ as slice_
  File "/usr/share/invesalius/invesalius/data/slice_.py", line 34, in <module>
    from invesalius.data.mask import Mask
  File "/usr/share/invesalius/invesalius/data/mask.py", line 35, in <module>
    from invesalius_cy import floodfill
ModuleNotFoundError: No module named 'invesalius_cy'
Traceback (most recent call last):
  File "/usr/lib64/python3.9/site-packages/wx/core.py", line 2326, in Notify
    self.notify()
  File "/usr/lib64/python3.9/site-packages/wx/core.py", line 3534, in Notify
    self.result = self.callable(*self.args, **self.kwargs)
  File "/usr/share/invesalius/app.py", line 131, in Startup2
    self.control = self.splash.control
AttributeError: 'Inv3SplashScreen' object has no attribute 'control'
Traceback (most recent call last):
  File "/usr/share/invesalius/app.py", line 261, in OnClose
    if self.fc.IsRunning():
AttributeError: 'Inv3SplashScreen' object has no attribute 'fc'
```

**Comment by benbullard79** [comment]
I get something different in ROME where it won't install:

```
# dnf in invesalius
Last metadata expiration check: 0:02:49 ago on Tue 27 Sep 2022 12:22:38 PM CDT.
Error:
 Problem: conflicting requests
  - package invesalius-3.1.99997-1.x86_64 requires python3dist(theano), but none of the providers can be installed
  - nothing provides python-dicom needed by invesalius-3.1.99993-1.x86_64
  - nothing provides python-itk needed by invesalius-3.1.99993-1.x86_64
  - nothing provides python-itk-numarray needed by invesalius-3.1.99993-1.x86_64
  - nothing provides python-sigar needed by invesalius-3.1.99993-1.x86_64
  - nothing provides python3.9dist(numpy) >= 1.9.1 needed by python-theano-1.0.5-1.x86_64
  - nothing provides python3.9dist(six) >= 1.9 needed by python-theano-1.0.5-1.x86_64
  - nothing provides python3.9dist(scipy) >= 0.14 needed by python-theano-1.0.5-1.x86_64
  - nothing provides python3.9dist(setuptools) needed by python-theano-1.0.5-1.x86_64
(try to add '--skip-broken' to skip uninstallable packages)

```

**Comment by benbullard79** [comment]
This issue is still present in cooker and rolling. At present the `inveslius` package builds with tests failed in cooker and rolling. This is what is known to be problem with `inveslius`:

```
DEBUG util.py:444:  Error: 
DEBUG util.py:444:   Problem: package invesalius-3.1.99998-2.x86_64 requires python3dist(scikit-image), but none of the providers can be installed
DEBUG util.py:444:    - conflicting requests
DEBUG util.py:444:    - nothing provides ((python3.9dist(matplotlib) < 3 or python3.9dist(matplotlib) > 3) with python3.9dist(matplotlib) >= 2) needed by python-scikit-image-0.18.3-1.x86_64
DEBUG util.py:444:    - nothing provides ((python3.9dist(pillow) < 7.1 or python3.9dist(pillow) > 7.1) with (python3.9dist(pillow) < 7.1.1 or python3.9dist(pillow) > 7.1.1) with python3.9dist(pillow) >= 4.3) needed by python-scikit-image-0.18.3-1.x86_64
DEBUG util.py:444:    - nothing provides python3.9dist(imageio) >= 2.3 needed by python-scikit-image-0.18.3-1.x86_64
DEBUG util.py:444:    - nothing provides python3.9dist(numpy) >= 1.16.5 needed by python-scikit-image-0.18.3-1.x86_64
DEBUG util.py:444:    - nothing provides python3.9dist(scipy) >= 1.0.1 needed by python-scikit-image-0.18.3-1.x86_64
DEBUG util.py:444:    - nothing provides python3.9dist(tifffile) >= 2019.7.26 needed by python-scikit-image-0.18.3-1.x86_64
DEBUG util.py:444:    - nothing provides python3.9dist(setuptools) needed by python-scikit-image-0.18.3-1.x86_64
DEBUG util.py:446:  (try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
DEBUG util.py:597:  Child return code was: 1

```
`python-scikit-image` says it won't build because:

`Fail reason | error: Bad exit status from /var/tmp/rpm-tmp.soKaOC (%build)`

Which does not tell us much. The logs for the x86_64 logs are [here](https://abf.openmandriva.org/build_lists/235519). I believe the logs for znver1 and aarch64 show same problem.  Maybe this section tells us why `python-scikit-image` is not happy:

```
DEBUG: Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.soKaOC
DEBUG: + umask 022
DEBUG: + cd /builddir/build/BUILD
DEBUG: + cd scikit-image-0.18.3
DEBUG: + '[' 1 -eq 1 ']'
DEBUG: + '[' 1 -eq 1 ']'
DEBUG: + '[' -e pyproject.toml ']'
DEBUG: + mkdir RPMBUILD_wheels
DEBUG: + CFLAGS='-Os -fomit-frame-pointer -g3 -gdwarf-4 -Wstrict-aliasing=2 -pipe -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fstack-protector-all --param=ssp-buffer-size=4 -m64 -mtune=generic -flto'
DEBUG: + pip wheel --wheel-dir RPMBUILD_wheels --no-deps --no-build-isolation --verbose .
DEBUG: /var/tmp/rpm-tmp.soKaOC: line 36: pip: command not found
DEBUG: error: Bad exit status from /var/tmp/rpm-tmp.soKaOC (%build)
DEBUG: RPM build errors:
DEBUG:     Bad exit status from /var/tmp/rpm-tmp.soKaOC (%build)
DEBUG: Child return code was: 1
INFO: EXCEPTION: [Error()]
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/mockbuild/trace_decorator.py", line 93, in trace
    result = func(*args, **kw)
             ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/mockbuild/util.py", line 599, in do_with_status
    raise exception.Error("Command failed: \n # %s\n%s" % (command, output), child.returncode)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mockbuild.exception.Error: Command failed: 
 # bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/python-scikit-image.spec
```

At any rate `python-scikit-image` needs to be fixed for `invesalius` to build. Unless there are still more dependency issues not yet discovered.

**Comment by rugyada** [comment]
Test in cooker - ISO build ID 1435 Live mode in VirtualBox

```
[live@omv-1435 ~]$ rpm -q invesalius
invesalius-3.1.99998-3.x86_64

*************************************
[live@omv-1435 ~]$ invesalius
[Errno 2] No such file or directory: '/home/live/.config/invesalius/config.json'
[Errno 2] No such file or directory: '/home/live/.config/invesalius/config.cfg'
Traceback (most recent call last):
  File "/usr/share/invesalius/app.py", line 117, in OnInit
    self.splash = Inv3SplashScreen()
                  ^^^^^^^^^^^^^^^^^^
  File "/usr/share/invesalius/app.py", line 161, in __init__
    dialog = lang_dlg.LanguageDialog()
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/invesalius/invesalius/gui/language_dialog.py", line 102, in __init__
    self.__TranslateMessage__()
  File "/usr/share/invesalius/invesalius/gui/language_dialog.py", line 186, in __TranslateMessage__
    _ = i18n.InstallLanguage("en")
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/invesalius/invesalius/i18n.py", line 81, in InstallLanguage
    lang = gettext.translation('invesalius', language_dir,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: translation() got an unexpected keyword argument 'codeset'
OnInit returned false, exiting...
```
Attached install log
[cooker-invesalius.log](https://github.com/OpenMandrivaAssociation/distribution/files/9794636/cooker-invesalius.log)

**Comment by omakuta** [comment]
This issue has been mentioned on **OMA forum**. There might be relevant details there:

https://forum.openmandriva.org/t/invesalius-aggiornato-per-cooker-rome-e-4-3/4678/6

**Comment by rugyada** [comment]
Fixed in invesalius-3.1.99998-4
